README.md

```diff
@@ -88,6 +88,66 @@ cat spec/fixtures/bb112zx3193.json | janeway "$.description.contributor[?@.role[
+
+For example, when displaying contributors, the label is usually determined by the role of the contributor, and the values are the display names of the contributors with that role:
+
+```ruby
+> rec.contributor_display_data.first.label
+=> "Former owner"
+> rec.contributor_display_data.first.values
+=> ["Hearst Magazines, Inc."]
+```
+
+The `#to_hash` helper method, which collapses all provided `DisplayData` into a single hash, can be used as a quick way to check the overall structure:
+
+```ruby
+> CocinaDisplay::DisplayData.to_hash(record.subject_display_data)
+=> {"Marque"=>["Bugatti"], "Model"=>["Bugatti T51A"], "Subject"=>["Bugatti automobile"]}
+```
+
+Note that usage of the `displayLabel` attribute in Cocina overrides the `label` that would ordinarily be used to group an item. It's possible for use of `displayLabel` to result in items being grouped together after relabeling, or to be split out into their own group.
+
+#### Creating display data
+
+In some cases, you may wish to create `DisplayData` for some data that isn't immediately available via a `_display_data` method. Depending on what you have, there are several helper methods available to do this that will label and group the data for you.
+
+If the data you have is an array of objects that respond to `#label` and `#to_s`, you can use `DisplayData.from_objects`, which will automatically group the objects by their `label` and set the `values` to the result of calling `#to_s` on each object. Most of the objects returned by `CocinaRecord` methods, like `Contributor` and `Subject`, respond to these methods.
+
+```ruby
+# This is actually equivalent to record.contributor_display_data!
+> CocinaDisplay::DisplayData.from_objects(record.contributors)
+```
+
+If the data you have is a simple array of strings, you can use `DisplayData.from_strings`. You need to provide a `label` to group the strings under:
+
+```ruby
+> CocinaDisplay::DisplayData.from_strings(["Bugatti", "Bugatti T51A", "Bugatti automobile"], label: "Subject")
+```
+
+If the data you have is a hash from parsed Cocina JSON, you can use `DisplayData.from_cocina`. You can optionally provide a `label` to use if the Cocina did not contain a `displayLabel` attribute:
+
+```ruby
+> cocina = { 'value' => 'Bugatti', 'displayLabel' => 'Marque' }
+# Will create a DisplayData with label "Marque" and value "Bugatti"
+> CocinaDisplay::DisplayData.from_cocina(cocina)
+# The same, but if the Cocina did not contain a displayLabel, it would use "Brand" instead
+> CocinaDisplay::DisplayData.from_cocina(cocina, label: 'Brand')
+```
+
+#### Custom formatting
+
+In some cases, you may need more control over the formatting. The `DisplayData#objects` method gives access to the underlying objects that were grouped under a particular `label`, allowing you to format them as needed.
+
+For contributors, the underlying objects are `Contributor` instances, which provide access to the associated `Name` and `Role` objects:
```

Comment:
I think especially for contributor it is nice to point out that the first and last name are available as separate values. We ran into needing that for schema.org markup so it is nice to know that most things are there you just have to look. Also pointing to where to look might be nice.